diff --git a/python-lib/cuddlefish/__init__.py b/python-lib/cuddlefish/__init__.py
--- a/python-lib/cuddlefish/__init__.py
+++ b/python-lib/cuddlefish/__init__.py
@@ -596,7 +596,7 @@ def run(arguments=sys.argv[1:], target_cfg=None, pkg_cfg=None,
     
     # reparse configs with arguments from local.json
     if config_args:
-        parser_kwargs['arguments'] += config_args
+        parser_kwargs['arguments'] = config_args + parser_kwargs['arguments']
         (options, args) = parse_args(**parser_kwargs)
 
     command = args[0]
