diff --git a/python-lib/mozrunner/killableprocess.py b/python-lib/mozrunner/killableprocess.py
--- a/python-lib/mozrunner/killableprocess.py
+++ b/python-lib/mozrunner/killableprocess.py
@@ -106,12 +106,23 @@ if not mswindows:
 class Popen(subprocess.Popen):
     kill_called = False
     if mswindows:
-        def _execute_child(self, args, executable, preexec_fn, close_fds,
-                           cwd, env, universal_newlines, startupinfo,
-                           creationflags, shell,
-                           p2cread, p2cwrite,
-                           c2pread, c2pwrite,
-                           errread, errwrite):
+        def _execute_child(self, *args_tuple):
+            if sys.hexversion < 0x02070600: # prior to 2.7.6
+                (args, executable, preexec_fn, close_fds,
+                    cwd, env, universal_newlines, startupinfo,
+                    creationflags, shell,
+                    p2cread, p2cwrite,
+                    c2pread, c2pwrite,
+                    errread, errwrite) = args_tuple
+                to_close = set()
+            else: # 2.7.6 and later
+                (args, executable, preexec_fn, close_fds,
+                    cwd, env, universal_newlines, startupinfo,
+                    creationflags, shell, to_close,
+                    p2cread, p2cwrite,
+                    c2pread, c2pwrite,
+                    errread, errwrite) = args_tuple
+
             if not isinstance(args, types.StringTypes):
                 args = subprocess.list2cmdline(args)
             
